### PR TITLE
Bring tables up to date with new classifier

### PR DESCRIPTION
**Pull request by snim2**
This PR updates the table scripts in two ways:

  1. to use the new "good" and "bad" inconsistent symbols
  2. to take account of the new classifier, which can now decide that a steady state has been reached *before* the last segment

The commits need quite a bit of squashing.

### Octane 0.7 test case

  * [Table as per master branch](https://dl.dropbox.com/s/t447w7wr2vsg3x3/master_branch_octane_table.pdf?dl=0)
  * [Table as per this PR](https://dl.dropbox.com/s/aelwfbnf6qn0yw7/new_table.pdf?dl=0)
  * [V8 plots](https:/dl.dropbox.com/s/4jmq3lgtmg1w5ic/octane_0_7_v8.pdf?dl=0)
  * [Spidermonkey plots](https://dl.dropbox.com/s/c6urdoui5x75xm6/octane_0_7_spidermonkey.pdf?dl=0)

### Bencher 3 warmup test case

  * [Table as per master branch](https://dl.dropbox.com/s/a7h3da1pw5bqaca/bencher3_master_branch_table.pdf?dl=0)
  * [Table as per this PR](https://dl.dropbox.com/s/vigoxoed4fh3n66/bencher3_new_table.pdf?dl=0)
  * [Plots](https://dl.dropbox.com/s/jqqolwcgpbimqvj/bencher3_plots.pdf?dl=0)

### Bencher 5 warmup test case

  * [Table as per master branch](https://dl.dropbox.com/s/z3c9ir6s3rpmxdf/bencher5_master_branch_table.pdf?dl=0)
  * [Table as per this PR](https://dldropbox.com/s/uj0qurol0cr3r6x/bencher5_new_table.pdf?dl=0)
  * [Plots](https://dl.dropbox.com/s/ot2ooxzkarxdzic/bencher5_plots.pdf?dl=0)

### Bencher 6 warmup test case

  * [Table as per master branch](https://dl.dropbox.com/s/rmdmgsrv4h3vx1p/bencher6_master_branch_table.pdf?dl=0)
  * [Table as per this PR](https://dl.dropbox.com/s/61shtjknoh1mimh/bencher6_new_table.pdf?dl=0)
  * [Plots](https://dl.dropbox.com/s/vj61gv0m8i6pc5p/bencher6_plots.pdf?dl=0)

### CODE REVIEW

File: bin/table_classification_summaries_others [hunk 1]
```diff
@@ -58,19 +60,39 @@ def main(data_dcts, window_size, latex_file, num_splits, with_preamble=False):
...
+                # its adjacent segments, so we first need to know which segments
+                # are steady-state segments.
+                first_steady_segment = len(data_dcts[machine]['changepoint_means'][key][p_exec]) - 1
+                num_steady_segments = 1
+                last_segment_mean = data_dcts[machine]['changepoint_means'][key][p_exec][-1]
+                lower_bound = min(last_segment_mean * (1.0 - half_bound), last_segment_mean - delta)
+                upper_bound = max(last_segment_mean * (1.0 + half_bound), last_segment_mean + delta)
+                for index in xrange(len(data_dcts[machine]['changepoint_means'][key][p_exec]) - 2, -1, -1):
+                    current_segment_mean = data_dcts[machine]['changepoint_means'][key][p_exec][index]
+                    if (current_segment_mean >= lower_bound and current_segment_mean <= upper_bound):
```
Comment (ltratt) [inline bin/table_classification_summaries_others]: Extraneous brackets?

File: bin/table_classification_summaries_others [hunk 2]
```diff
@@ -82,31 +104,37 @@ def main(data_dcts, window_size, latex_file, num_splits, with_preamble=False):
                 # Consistent benchmark, but some process execs failed with error.
                 reported_category = ' %s\\scriptsize{($%d$)}' % \
                                     (STYLE_SYMBOLS[categories[0]], len(categories))
-            else:
-                reported_category = STYLE_SYMBOLS['inconsistent']
+            elif set(categories) == set(['flat', 'warmup']):
+                reported_category = STYLE_SYMBOLS['good inconsistent']
+                cat_counts = list()
```
Comment (ltratt) [inline bin/table_classification_summaries_others:109]: Any reason not to use the shorter `cat_counts = []`?

**Review by ltratt — APPROVED** [review]

File: bin/table_classification_summaries_others (hunk 2 above)
Comment (snim2) [reply bin/table_classification_summaries_others:109]: consistency with the other scripts in the repo

**Comment by snim2** [comment]
Table with new symbols: [new_table.pdf](https://github.com/softdevteam/warmup_experiment/files/784377/new_table.pdf)

**Comment by ltratt** [comment]
This looks good. One tweak that I think we should add now is fixing the order of inconsistent sub-things. At the moment you can have "inconsistent (flat, warmup, nss)" or "inconsistent(warmup, nss, flat)" or any combination. I don't mind what order they're in, provided they're consistent.

**Comment by snim2** [comment]
They are in order of number of occurrences, so that you see the most common case first when reading left->right.

**Comment by ltratt** [comment]
Ah, point taken.

OK, so shall we squash?

**Comment by snim2** [comment]
OK, squashed!

**Pull request merged**